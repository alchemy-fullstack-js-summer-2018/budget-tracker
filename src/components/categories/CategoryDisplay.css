:local(.categoryDisplay) {
  .alert {
    color: red;
  }
}