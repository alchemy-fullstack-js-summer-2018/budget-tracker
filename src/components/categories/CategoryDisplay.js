import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { remove } from './actions';

class CategoryDisplay extends Component {
  static propTypes = {
    category: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    remove: PropTypes.func.isRequired
  };

  render() {
    const { category, onEdit, remove } = this.props;

    return (
      <p>
        {category.name} allowed {category.budget}
        <button name="edit" onClick={onEdit}>âœŽ</button>
        <button id="delbtn" name="delete" onClick={() => remove(category.key)}>ðŸ—‘</button>     
      </p>
    );
  }
}

export default connect (
  null,
  { remove }
)(CategoryDisplay);