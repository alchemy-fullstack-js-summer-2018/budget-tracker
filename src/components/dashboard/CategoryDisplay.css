:local(.categoryDisplay) {
  text-align: center;
}