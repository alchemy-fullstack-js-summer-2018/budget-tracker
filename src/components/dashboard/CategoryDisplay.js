import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { remove } from './actions';

class CategoryDisplay extends Component {
  
  static propTypes = {
    category: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    remove: PropTypes.func.isRequired
  };

  render() {
    const { category, onEdit, remove } = this.props;

    return (
      <Fragment>
        <p>
          <strong>Category:</strong> {category.name}<br/>
          <strong>Budget:</strong> ${category.budget}<br/>
          <button name="edit" onClick={onEdit}>Edit</button>
          <button name="delete" onClick={() => remove(category.id)}>Delete</button>
        </p>
      </Fragment>
    );
  }
}

export default connect(
  null,
  { remove }
)(CategoryDisplay);