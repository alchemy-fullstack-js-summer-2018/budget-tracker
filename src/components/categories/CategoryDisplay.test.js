import { CategoryDisplay } from './CategoryDisplay';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Category item display', () => {
  it('renders category display', () => {
    const handleEdit = jest.fn();
    const remove = jest.fn();

    const wrapper = shallow(<CategoryDisplay
      category={{ name: 'name', budget: 1 }}
      onEdit={handleEdit}
      remove={remove}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.find('button[name="edit"]').simulate('click');
    const editCalls = handleEdit.mock.calls;
    expect(editCalls.length).toBe(1);

    wrapper.find('button[name="delete"]').simulate('click');
    const removeCalls = remove.mock.calls;
    expect(removeCalls.length).toBe(1);
  });
});