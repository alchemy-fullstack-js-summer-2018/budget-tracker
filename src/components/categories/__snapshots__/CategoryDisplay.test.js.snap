// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category display renders display, then edit 1`] = `
<div>
  <p>
    Category: 
    Rent
  </p>
  <p>
    Budget: $
    500
  </p>
  <p>
    You have spent: $
  </p>
  <p>
    Created: 
    2095-12-17T03:24:00
  </p>
  <button
    name="edit"
    onClick={[MockFunction]}
  >
    ✎
  </button>
  <button
    name="delete"
    onClick={[Function]}
  >
    ❌
  </button>
  <Connect(Expenses)
    categoryId="PPBqWA9"
  />
</div>
`;
